src/python_testing/TC_XX_1_1.py

```diff
@@ -0,0 +1,138 @@
+#
+#    Copyright (c) 2022 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import logging
+
+from matter_testing_support import MatterBaseTest, async_test_body, default_matter_test_main
+from mobly import asserts
+
+standard_range_prefix = 0x0000
+manufacturer_code_min_prefix = 0x0001
+manufacturer_code_max_prefix = 0xFFF0
+test_vendor_min_prefix = 0xFFF1
+test_vendor_max_prefix = 0xFFF4
+invalid_vendor_min_prefix = 0xFFF5
+invalid_vendor_max_prefix = 0xFFFF
+
+attribute_id_range_min = 0x0000
+attribute_id_range_max = 0x4FFF
+attribute_global_range_min = 0xF000
+attribute_global_range_max = 0xFFFE
+attribute_id_invalid_max = 0xFFFF
+
+# event_id_range_min = 0x00
+# event_id_range_max = 0xFF
+# event_id_invalid_max = 0xFFFF
+
+command_id_range_min = 0x00
+command_id_range_max = 0xFF
+command_id_invalid_max = 0xFFFF
+
+ATTRIBUTE_LIST_ID = 0xFFFB
+# EVENT_LIST_ID = 0xFFFA
+ACCEPTED_COMMAND_LIST_ID = 0xFFF9
+GENERATED_COMMAND_LIST_ID = 0xFFF8
+
+
+def get_element_prefix_suffix(element):
+    prefix_mask = ~0xffff
+    suffix_mask = 0xffff
+
+    prefix = (element & prefix_mask) >> 16
+    suffix = (element & suffix_mask)
+
+    return prefix, suffix
+
+
+def is_attribute_id_in_valid_range(attribute_id):
+    prefix, suffix = get_element_prefix_suffix(attribute_id)
+
+    if prefix == standard_range_prefix:
+        if suffix > attribute_id_range_max and suffix < attribute_global_range_min or suffix == attribute_id_invalid_max:
+            asserts.fail(f"Invalid attribute id {attribute_id:010x} in standard range")
+
+    elif prefix >= manufacturer_code_min_prefix and prefix <= manufacturer_code_max_prefix:
+        if suffix > attribute_id_range_max:
+            asserts.fail(f"Invalid attribute id {attribute_id:010x} in MC range")
+
+    elif prefix >= test_vendor_min_prefix and prefix <= test_vendor_max_prefix:
+        if suffix >= attribute_id_range_min:
```

Comment:
This looks like it fails for all attribute ids, but in that case, why are we checking the suffix at all?  This at least needs a comment describing the intended behavior...

Reply:
Yeah, I get your point. If the complete range is disallowed, there is no need to hide it being what looks like a range check. I will update this.

Reply:
For the test vendor ID, it was discussed to not outright fail since the CI/SDK uses this for some testing elements, so I will try to update the script to be inline with this.

The test plan should also have been updated to change this section to a SHOULD NOT statement.

Reply:
This is addressed in the latest commit